### PR TITLE
[sonataflow-operator] - Migrate SonataFlow operator from incubator-kie-kogito-serverless-operator

### LINKED ISSUE #2605: SonataFlow Operator migration to `kie-tools`
Task to migrate https://github.com/apache/incubator-kie-kogito-serverless-operator to https://github.com/apache/incubator-kie-tools. Details can be seen on #2603.

### PR DESCRIPTION
**Pull request by ricardozanini**
Fix #2605 

In this PR:

1. Upgrade repo Go version to 1.22
2. Remove the old `sonataflow-operator` for the 10.0.0 release
3. Push the new `sonataflow-operator` from https://github.com/apache/incubator-kie-kogito-serverless-operator repository

TODO:

- [x] Remove unrelated commits from the bddframework module
- [x] Upgrade CI to Golang 1.22

### CODE REVIEW

**Comment by ricardozanini** [comment]
I will fix the windows problem tomorrow.

**Review by tiagobento — APPROVED** [review]
Thanks for the updates @ricardozanini! Good to go when green. I especially like the stats ;)

image

**Comment by tiagobento** [comment]
Mental note for not merging it before squashing the new commits.

**Comment by tiagobento** [comment]
BIG Mental note for **NOT SQUASHING** this PR when merging.

**Comment by ricardozanini** [comment]
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**
**DO NOT SQUASH**

**Comment by ricardozanini** [comment]
@wmedvede @treblereel @jakubschwan wanna take a look?

File: packages/kn-plugin-workflow/README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ All the commands in this section should be performed in the monorepo root.
 
 - Node `>= 20.14.0` _(To install, follow these instructions: https://nodejs.org/en/download/package-manager/)_
 - pnpm `9.3.0` _(To install, follow these instructions: https://pnpm.io/installation)_
-- Go `1.21.13` _(To install, follow these instructions: https://go.dev/doc/install)_
+- Go `1.22.9` _(To install, follow these instructions: https://go.dev/doc/install)_
```
Comment (treblereel) [inline packages/kn-plugin-workflow/README.md:32]: Maybe it should be go1.23.2 like in the go.mod?

**Review by treblereel — APPROVED** [review]
looks good for me

File: packages/kn-plugin-workflow/README.md (hunk 1 above)
Comment (tiagobento) [reply packages/kn-plugin-workflow/README.md:32]: From what @ricardozanini explained to me, 1.23.2 is only being used for Go's toolchain, which doesn't affect compilation. He'll see if it's worth aligning everything in a future PR.

**Pull request merged**